pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go

```diff
@@ -86,7 +96,41 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, imc *v1.InMemoryChannel)
 		}
 	}
 	return nil
+}
+
+func (r *Reconciler) patchSubscriberStatus(ctx context.Context, imc *v1.InMemoryChannel) error {
+	after := imc.DeepCopy()
 
+	after.Status.Subscribers = make([]eventingduckv1.SubscriberStatus, 0)
+	for _, sub := range imc.Spec.Subscribers {
+		after.Status.Subscribers = append(after.Status.Subscribers, eventingduckv1.SubscriberStatus{
+			UID:                sub.UID,
+			ObservedGeneration: sub.Generation,
+			Ready:              corev1.ConditionTrue,
+		})
+	}
+	jsonPatch, err := duck.CreatePatch(imc, after)
+	if err != nil {
+		logging.FromContext(ctx).Warnw("Failed to create JSON patch", zap.Error(err))
+		return err
+	}
+	patch, err := jsonPatch.MarshalJSON()
+	if err != nil {
+		logging.FromContext(ctx).Warnw("Failed to marshal JSON patch", zap.Error(err))
+		return err
+	}
+	// If there is nothing to patch, we are good, just return.
+	// Empty patch is {}, hence we check for that.
+	if len(patch) <= 2 {
+		return nil
+	}
```

Comment:
Is it something we could possibly check _before_ marshaling `jsonPatch`, by implementing `func (JSONPatch) Empty() bool` for example?

Reply:
good idea!